.github/workflows/poetry-check-sdist.yml

```diff
@@ -6,8 +6,8 @@ jobs:
     strategy:
       fail-fast: false
       matrix:
-        python-version: [3.7, 3.8]
```

Comment:
the purpose of this build is to check that the build sdist can correctly be installed.

it would be nice to be sure that both the sdist and the wheel that we distribute can be installed correctly, as the problems you have had were introduced without breaking the build.

Reply:
I think this issue is now being picked up by the CI. For example, in your PR #59 from yesterday, the sdist builds are failing with the same error I saw on google colab:

https://github.com/jwg4/exact_cover/actions/runs/4088835028/jobs/7050902881

```
ERROR tests/test_exact_cover.py - ModuleNotFoundError: No module named 'exact_cover_impl'
```
I think the sdist build problems stem from a breaking change in poetry-core
https://github.com/python-poetry/poetry-core/pull/318
which as of last week's release of poetry-core 1.5.0
https://github.com/python-poetry/poetry-core/releases
no longer generates the setup.py needed for building the binary extension module without us explicitly setting `generate-setup-file` to true (which is what this PR does, and why the tests are passing in this PR but not PR #59).

The breaking change to poetry-core was only released a week ago, which is why it's only messing up the CI now. In the comments of the poetry PR, one person comments "I am fairly certain releasing this will break for a few folks building extensions using setuptools, just not sure how large the number is going to be." I think it has indeed broken the exact-cover package, but this PR fixes it.